config/config_params.go

```diff
@@ -143,7 +151,7 @@ type Config struct {
 	OpenstackRegion string `config:"region;;die-on-fail"`
 
 	InterfacePrefix  string `config:"iface-list;cali;non-zero,die-on-fail"`
-	InterfaceExclude string `config:"iface-list;kube-ipvs0"`
+	InterfaceExclude string `config:"iface-list-regexp;kube-ipvs0;die-on-fail"`
```

Comment:
Calling this out, since it changes the default behaviour. 

Added the `die-on-fail` option, because it seemed sensible for it to halt on failed validation (it's going to panic anyways during the `Config.InterfaceExcludes()` if any of the individual values do not compile).

Reply:
I'm not sure I'm fully understanding here, but it feels wrong for this detail of behaviour to change.  I think it's straightforward to avoid "it's going to panic anyways ... if any of the individual values do not compile" by using `regexp.Compile` in that code instead of `regexp.MustCompile`.

Reply:
(die-on-fail will cause Felix to go into a restart loop, so we generally specify die-on-fail only for essential config parameters)

Reply:
> I think it's straightforward to avoid "it's going to panic anyways ... if any of the individual values do not compile" by using `regexp.Compile` in that code instead of `regexp.MustCompile`.

Right. 

When I originally considered this I came to a conclusion that using `regexp.Compile` to avoid a panic would require knowing what we want to do when the one of the user provided values fails to compile to regexp. What should we do in that case? We can log the error message as a bare minimum. But if we gracefully continue onwards, it means we probably don't setup that particular interface exclude value. I just wasn't sure from a user experience standpoint what our acceptable behaviour was. E.g. if I wanted to exclude all interfaces with a name starting with `kube` but I somehow mess up the syntax such that it doesn't compile. Ideally I would want `felix` to catch my mistake and halt during startup to tell me so that I can correct myself. 

So that was my original intention. 

> (die-on-fail will cause Felix to go into a restart loop, so we generally specify die-on-fail only for essential config parameters)

 Fair point. I will remove the `die-on-fail` and switch the compiling of values from `regexp.MustCompile` to `regexp.Compile`. 👍

Reply:
Actually looking at the logic for handling `die-on-fail` (`DieOnParseFailure` flag) after parsing a parameter it looks like it wipes the error and then sets the value to default, which for `InterfaceExclude` is `kube-ipvs0`. That seems undesirable (especially now that we could have regex values), no? 

https://github.com/projectcalico/felix/blob/master/config/config_params.go#L335

Reply:
Yes, the behaviour for a parse error for a parameter without `die-on-fail` is to log a warning and then use the default value.  I think that's what we should do here too.

Reply:
Okay, sounds good. I thought it over again and you're right, there's no reason to change this behaviour from what it's always done. That's what users would expect since it's established. 

Changing it back.